src/main/java/oof/Storage.java

```diff
@@ -161,6 +165,61 @@ public void writeTaskList(TaskList taskList) {
         throw new IOException();
     }
 
+    /**
+     * Writes Tracker objects to hard disk.
+     * @param trackerList   TrackerList of Tracker objects.
+     * @throws OofException if unable to write TrackerList.
+     */
+    public void writeTrackerList(TrackerList trackerList) throws OofException {
+        try {
+            List<oof.model.tracker.Tracker> arr = trackerList.getTrackers();
+            String filename = PATH_TRACKER;
+            BufferedWriter out = new BufferedWriter(new FileWriter(filename));
+            for (oof.model.tracker.Tracker tracker : arr) {
+                out.write(tracker.toStorageString() + "\n");
+            }
+            out.close();
+        } catch (IOException e) {
+            throw new OofException("Unable to save Tracker data.");
+        }
+    }
+
+    /**
+     * Reads Tracker objects that were previously saved to hard disk.
+     * @return TrackerList that contains Tracker objects.
+     * @throws OofException if unable to read tracker.txt.
+     */
+    public ArrayList<Tracker> readTrackerList() throws OofException {
+        try {
+            File file = new File(PATH_TRACKER);
+            FileReader fileReader = new FileReader(file);
+            BufferedReader br = new BufferedReader(fileReader);
+            ArrayList<oof.model.tracker.Tracker> trackerList = new ArrayList<>();
```

Comment:
do not use such conventions.

Reply:
This issue has been rectified as suggested.